tools/isobuild/meteor-npm.js

```diff
@@ -528,10 +531,10 @@ const isPortable = Profile("meteorNpm.isPortable", dir => {
 
   const result = hasBuildScript
     ? false // Build scripts may not be portable.
-    : optimisticReaddir(dir).every(
+    : optimisticReaddir(dir, { withFileTypes: true }).every(
```

Comment:
`optimisticReaddir` doesn't seem to watch the contents of the directory. Before, only the list of names could become stale, and now the types can become stale also.

Reply:
Optimistic functions don't cache when an argument is an object, so my above comment is partly incorrect. I will need to check if this causes performance regressions in some cases.